src/transform/src/dataflow.rs

```diff
@@ -94,8 +98,9 @@ fn optimize_dataflow_filters(dataflow: &mut DataflowDesc) {
                 .as_mut()
                 .take_dangerous()
                 .filter(list.iter().cloned());
+            // TODO: remove the predicates in `list` from the earlier objects
```

Comment:
So I think this `TODO` is not right, with the current framing elsewhere surrounding `LinearOperator`. At the moment, and we could certainly change this, it represents *strictly optional* transformations that can be applied. We haven't done enough yet to be certain it is applied in all cases, and so until the contract changes, we shouldn't remove any existing filters. There is a very reasonable `TODO` that reads like "change the contract to *ensure* that filtering, projection (and possibly mapping) is applied, so that projections and filters can be removed", but I'd like to do that as a separate PR.